exec/selected/selected.go

```diff
@@ -55,6 +55,60 @@ type SchemaField struct {
 	FixedResult reflect.Value
 }
 
+func (sf *SchemaField) ToSelectedField() *types.SelectedField {
+	if sf == nil {
+		return nil
+	}
+	return &types.SelectedField{
+		Name:       sf.Name,
+		Alias:      sf.Alias,
+		Fields:     selsToSelectedFields(sf.Sels),
+		Args:       sf.Args,
+		Directives: sf.Directives,
+	}
+}
+
+func selsToSelectedFields(sels []Selection) (fs []*types.SelectedField) {
+	for _, f := range sels {
+		// Selection is SchemaField or TypeAssertion or TypenameField
+		switch v := f.(type) {
+		case *SchemaField:
+			fs = append(fs, v.ToSelectedField())
+		case *TypeAssertion:
+			/*
+				TypeAssertion is poorly commented so let's do it here - it represents selections on union fields
+				# Example
+				## Schema
+				type A {}
+				type B {}
+				union AB = A | B
+				type Test {
+					ab: AB
+				}
+				## Query
+				query {
+					test {
+						ab {
+							...on A { # represented by TypeAssertion
+								?
+							}
+							...on B { # represented by TypeAssertion
+								?
+							}
+						}
+					}
+				}
+			*/
+
+			fs = append(fs, selsToSelectedFields(v.Sels)...)
```

Comment:
is this what we want to do with them??

we need support for type assertions in the exported struct